### PR TITLE
Implement .indices(len) of slice

### LINKED ISSUE #1431: Implement slice method
## Feature
Currently, the `__repr__`, comparison, and` indices` functions are not implemented in slice.

### Expected result
```python
>>> slice(10)
slice(None, 10, None)
>>> slice(10).indices(0)
(0, 0, 1)
>>> slice(1) < slice(20)
True
>>> slice(10) == slice(10)
True
```

### Actual result
```python
>>>>> slice(10)
<slice object at 0x7fd391659260>
>>>>> slice(10).indices()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'slice' object has no attribute 'indices'
>>>>> slice(1) < slice(20)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unsupported operand types for '<': 'slice' and 'slice'
>>>>> slice(10) == slice(10)
False
```

### PR DESCRIPTION
**Pull request by tcmal**
Fixes #1431.

CPython equivalent method: https://github.com/python/cpython/blob/master/Objects/sliceobject.c#L373

### CODE REVIEW

**Comment by dralley** [comment]
Since the code here uses the same algorithm, we should call this function from here as well.  https://github.com/RustPython/RustPython/blob/master/vm/src/obj/objrange.rs#L317-L383

File: vm/src/obj/objrange.rs [hunk 1]
```diff
@@ -319,78 +319,12 @@ impl PyRange {
...
-                        upper_bound.clone()
-                    } else {
-                        slice_stop.clone()
-                    }
-                } else if negative_step {
-                    lower_bound.clone()
-                } else {
-                    upper_bound.clone()
-                };
-
```
Comment (dralley) [inline vm/src/obj/objrange.rs]: The 3 lines below do need to remain here, though.

Comment (dralley) [reply vm/src/obj/objrange.rs]: The function is equivalent to this CPython code (but the variables are named opposite) https://github.com/python/cpython/blob/master/Objects/rangeobject.c#L306-L322

**Comment by windelbouwman** [comment]
@tcmal could you add a python test snippet with an example of the problem you fixed? This would help to prevent the issue from re-appearing.

File: tests/snippets/attr.py [hunk 2]
```diff
@@ -4,12 +4,21 @@
 class A:
     pass
 
+class B:
+    x = 50
```
Comment (windelbouwman) [inline tests/snippets/attr.py]: This looks like code from another pull request, is it related to this change, or is it due to git merging?

Comment (tcmal) [reply tests/snippets/attr.py]: Yeah messed up the rebase

**Comment by tcmal** [comment]
Added tests for slice, mostly based off [CPython's code](https://github.com/python/cpython/blob/master/Lib/test/test_slice.py).

**Comment by dralley** [comment]
Is this waiting on anything?

**Comment by dralley** [comment]
@windelbouwman

**Pull request merged**